app/src/battery.c

```diff
@@ -27,7 +27,13 @@ uint8_t zmk_battery_state_of_charge() { return last_state_of_charge; }
 static int zmk_battery_update(const struct device *battery) {
     struct sensor_value state_of_charge;
 
+#if defined(CONFIG_ZMK_BATTERY_FETCH_ALL)
+    int rc = sensor_sample_fetch(battery);
+#elif defined(ZMK_BATTERY_FETCH_INDIVIDUAL_CHANNELS)
     int rc = sensor_sample_fetch_chan(battery, SENSOR_CHAN_GAUGE_STATE_OF_CHARGE);
+#else
+#error "No value selected for ZMK_BATTERY_FETCH"
+#endif
```

Comment:
Because the default w/ this change is "fetch all", this changes the default fetch style, which potentially risks some breakage for any drivers that don't support that. Thoughts on having "individual channels" be the default choice instead?

Reply:
Done.